core/git-access.c

```diff
@@ -743,63 +741,62 @@ int sync_with_remote(git_repository *repo, const char *remote, const char *branc
 		if (is_subsurface_cloud)
 			report_error("Cannot sync with cloud server, working with offline copy");
 		else
-			report_error("Unable to fetch remote '%s'", remote);
+			report_error("Unable to fetch remote '%s'", info->url);
 		// If we returned GIT_EUSER during authentication, giterr_last() returns NULL
 		SSRF_INFO("git storage: remote fetch failed (%s)\n", giterr_last() ? giterr_last()->message : "authentication failed");
 		// Since we failed to sync with online repository, enter offline mode
 		git_local_only = true;
 		error = 0;
 	} else {
-		error = check_remote_status(repo, origin, remote, branch, rt);
+		error = check_remote_status(info, origin);
 	}
 	git_remote_free(origin);
 	git_storage_update_progress(translate("gettextFromC", "Done syncing with cloud storage"));
 	return error;
 }
 
-static git_repository *update_local_repo(const char *localdir, const char *remote, const char *branch, enum remote_transport rt)
+static bool update_local_repo(struct git_info *info)
 {
 	int error;
-	git_repository *repo = NULL;
```

Comment:
I haven't walked back to the callers to understand this better... but here we used to initialize the `repo` as `NULL, but we don't initialize `info->repo` as `NULL`... is that intentional?

Reply:
We initialize everything to zero with

         memset(info, 0, sizeof(*info));

in is_git_repository().